### PR TITLE
Show toolbar and follow button for list visitor

### LINKED ISSUE #9337: Fix List pages so MyBooks sidebar + Follows btn appear
Thank you @Realmbird -- can you see if you can add a "Follow" button to the **List** pages?

When a visitor goes to a patron's **Lists** page, they see a follow button. 
Screenshot 2024-05-24 at 11 17 24 AM

Here's an example of all a patron's lists, that show follow btn:
https://openlibrary.org/people/mekBot/lists

This is handled by the code in:
https://github.com/internetarchive/openlibrary/blob/master/openlibrary/templates/account/view.html

However, if a visitor goes to a **specific** patron's list (i.e. /people/xxx), then the follow button disappears, as does the my books sidebar and we want (a) the sidebar to appear and (b) the follow button to remain.

Screenshot 2024-05-24 at 11 17 05 AM

Here's an example of a specific list: https://openlibrary.org/people/mekBot/lists/OL104041L/Popular_Books

Note: This already **does** the right thing, when a patron is logged in and goes to their own List:
Screenshot 2024-05-24 at 11 29 12 AM

The template we'll need to worry about for the single list case is:
https://github.com/internetarchive/openlibrary/blob/master/openlibrary/templates/type/list/view.html#L11-L20

Instead of `if` `else` blocks, we always want to return https://github.com/internetarchive/openlibrary/blob/master/openlibrary/templates/type/list/view.html#L12-L17 except the value of `is_owner` should be determined by the `is_owner` variable rather than True

### PR DESCRIPTION
**Pull request by schu96**
Closes #9337 

Fix

### Technical

Remove ```if/else``` condition, uses the ```lst['key']``` to determine if the user is the owner of the list and show relevant components within the ```account/view.html``` template.

### Testing

- [Create two user accounts](https://github.com/internetarchive/openlibrary/wiki/Creating-and-Logging-in-as-a-new-user-on-your-local-client) and use one account to create a list of books
- Navigate to the list page and check that the follow button does not appear within the list of lists and single list webpages.
- Sign out of the list creator account and sign into the other newly created account.
- Navigate to the created book list, verify that the follow button appears on the list page and that pressing follow will add the creator's account to the visitor's ```Following``` tab.

### Screenshot

image
image
image
image

### Stakeholders

@jimchamp

### CODE REVIEW

File: openlibrary/templates/account/view.html [hunk 1]
```diff
@@ -52,7 +52,7 @@ <h1 class="details-title">$header_title</h1>
               $ share_links = ctx.get('share_links')
               $:macros.ShareModal(share_links, page_url, share_content, "share-modal-wrapper", show_embed=False)
         </div>
-      $if page:
+      $if page and mb.is_my_page:
```
Comment (jimchamp) [inline openlibrary/templates/account/view.html:55]:
```suggestion
      $if page:
```
Ensure that `databarView` always appears in the top-right corner of the view.

File: openlibrary/templates/account/view.html [hunk 2]
```diff
@@ -52,7 +52,7 @@ <h1 class="details-title">$header_title</h1>
...
         $:macros.databarView(page)
```
Comment (jimchamp) [inline openlibrary/templates/account/view.html:56]:
```suggestion
        $:macros.databarView(page, edit=mb.is_my_page)
```
Only show the edit button to the list's owner.

File: openlibrary/templates/type/list/view.html [hunk 3]
```diff
@@ -8,13 +8,9 @@
...
+$ template = MyBooksTemplate(lst['key'].split('/')[2], 'list').render(
+    $   header_title= "",
+    $   template=render_template("type/list/view_body", lst, is_owner),
+    $   page=lst
     $ )
-    $:template
-
-$else:
-    $:render_template("type/list/view_body", lst, is_owner)
+$:template
```
Comment (jimchamp) [inline openlibrary/templates/type/list/view.html:16]:
```suggestion
$:template

```
Add trailing newline to this file.

**Review by jimchamp — CHANGES_REQUESTED** [review]
Thanks @schu96!  This is working well, but there are two things that are bothering me:

Screenshot 2024-10-01 105524

1. There are two sets of navigation breadcrumbs.
2. The `databarView` macro doesn't appear in the top-right corner of the view.

I've left some suggested code changes that will partially address number 2.  In addition to those changes, you'll have to remove the breadcrumbs and the duplicate `databarView`, [here](https://github.com/internetarchive/openlibrary/blob/9eb111cc9514c27ff6aaa18b75b80f108e129c74/openlibrary/templates/type/list/view_body.html#L41-L47).

**Review by jimchamp — CHANGES_REQUESTED** [review]
So far, so good.  The duplicate breadcrumbs and `databarView` are [here](https://github.com/internetarchive/openlibrary/blob/9eb111cc9514c27ff6aaa18b75b80f108e129c74/openlibrary/templates/type/list/view_body.html#L41-L47), and should be removed.

**Comment by schu96** [comment]
Whoops, thanks for the reminder I had a complete airhead moment there

File: openlibrary/templates/account/view.html (hunk 1 above)
Comment (jimchamp) [reply openlibrary/templates/account/view.html]: I was totally wrong about this!  The header is too crammed when `databarView` is present for visitors.

**Review by jimchamp — APPROVED** [review]
Thanks @schu96, and sorry for the massive delay...
This is great!  I'm going to switch things back such that the `databarView` isn't displayed for visitors, as there's not enough room for it and people can check the history table if they want to see the last edit date.
Will merge after I make the change and the tests pass.

**Pull request merged**